src/lib/homestore_backend/hs_homeobject.cpp

```diff
@@ -272,6 +259,28 @@ void HSHomeObject::register_homestore_metablk_callback() {
             LOGI("Found existing SvcId: [{}]", to_string(_our_id));
         },
         nullptr, true);
+
+    // recover PG
+    HomeStore::instance()->meta_service().register_handler(
+        _pg_meta_name,
+        [this](homestore::meta_blk* mblk, sisl::byte_view buf, size_t size) {
+            on_pg_meta_blk_found(std::move(buf), voidptr_cast(mblk));
+        },
+        nullptr, true);
+
```

Comment:
`HomeStore::instance()->meta_service().read_sub_sb(_shard_meta_name);`
`HomeStore::instance()->meta_service().read_sub_sb(_pg_meta_name);`

don`t we need this ?

Reply:
yeah I also dont under stand how this works.  The initiative probably want to move these line before HS.start(), which should be put into `register_homestore_metablk_callback`

Reply:
I think we need this if we register callback after metablk init. If we move the register to before metablk init, the cb gets called